CHANGELOG.md

```diff
@@ -26,6 +26,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - User guide now explains the new objective classes
```

Comment:
@AdrianSosic 
would `rhs = field(converter= DTypeFloatTorch)` also work?

Reply:
this is the type botorch expects, the last tuple entry being the `rhs`, so Im wondering whether the tensor suggestions would even work 
![image](https://github.com/emdgroup/baybe/assets/17951239/be89f92c-d295-45da-b6b9-6d8c10580701)

Reply:
I've now tried these but theys till fail the tests:
`DTypeFloatNumpy(self.rhs)` and `torch.tensor(self.rhs, dtype=DTypeFloatTorch).item()`

going via attrs converters is not going well with the lazy loading

your suggestions makes it a tensor but the type hint expects a float

Reply:
> would `rhs = field(converter= DTypeFloatTorch)` also work?

No, and not surprised either because this is what I tried to describe in my original message – roundtrip serialization could easily fail, hence I'd refrain of modifying the user input here. 

> DTypeFloatNumpy(self.rhs) and torch.tensor(self.rhs, dtype=DTypeFloatTorch).item()
Also thought about the latter but the problem is that there is no such thing as a pure float32, afaik, which the last one tries to create.

I think the "correct" approach (and probably we should have done this way earlier instead of explicitly passing the dtype everywhere) is to simply set `torch.set_default_dtype(DTypeFloatTorch)` at some appropriate place. My hope is that this would automatically turn the `rhs` (provided as float) into the right dtype upon transformation to botorch. Could you give it a try?

Reply:
Here the errors Ig et (still with `torch.set_default_type`)
![image](https://github.com/emdgroup/baybe/assets/17951239/77e8552a-fb66-40bd-9ddd-b2b34f3759c7)

There are several types of sisues
1) Hypothesis creating breaking floats -> fixed by setting the correct `width`, only required if we want all tests to succeed in single precision
2) numerical instability -> no easy fix and perhaps also no required fix
3) Single-Double problem appearing with constraints -> none of the suggestions have fixed it so far including using `set_default_dtype` in the global package  `__init__` (didnt care about lazy import for this test)

I didnt investigate it yet but I could imagine that types are not cleanly handled by botorch side, after all the type hint for `rhs` is the Python `float` which as far as I know is always 64, or not?

Reply:
Let's quickly investigate where the 64-bit tensor is created. If on botorch side, we can raise an issue. And yes, AFAIK native python float is always 64-bit. We could probably still work around it by using np.float32

Reply:
@AdrianSosic seems botorch ehre does an explicit converison of `rhs` to `float`
![image](https://github.com/emdgroup/baybe/assets/17951239/87074229-4431-4b07-ac17-a078940afe3c)

also this referenced function has float64 hardcoded:
![image](https://github.com/emdgroup/baybe/assets/17951239/08800654-bbf8-4f26-91f6-2b7d54a6b8d0)

Reply:
Well, I guess there we have the culprit... So, what do you think? Open botorch issue? Last time, they fixed it rather quickly, indeed. But we do need to keep in mind that we'll get the change only with the next release...

Reply:
we should merge this PR and add an admonition in the userguide (I can add it to #227 ) saying that currently conti constraints do not support singlepoint precision
+ botoch issue and remove admonition should it get fixed and included by us

Reply:
sounds like a good plan to me 👍🏼 can you also please add a TODO in the code so that it shows up in the autogenerated TODO list?

Once this is done and you resolve the thread here, I'll have one more look at the PR and can then (hopefully) accept

Reply:
added TODO
added admonition in #227